### PR TITLE
feat: Add UWP DataGrid sample

### PR DESCRIPTION
**Pull request by morning4coffe-dev**
GitHub Issue (If applicable): #691

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the new behavior?

Gallery now showcases the features of DataGrid.
image

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested on iOS.
- [x] Tested on Wasm.
- [x] Tested on Android.
- [ ] Tested on UWP.
- [x] Tested in both **Light** and **Dark** themes.
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by morning4coffe-dev** [comment]
@MartinZikmund Here is my draft for the DataGrid sample. Can you look into the UWP regression please? Just to sum up what we discussed on our call, all Uno targets work correctly, but UWP throws `ERROR HRESULT E_FAIL has been returned from a call to a COM component`.

**Comment by morning4coffe-dev** [comment]
@MartinZikmund I have managed to fix the issue causing the mentioned Error. Binding a value to `RowGroupHeaderPropertyNameAlternative` seems to have caused this issue (it does this only on UWP not even in my standalone sample in WinUI).

File: Uno.Gallery/Uno.Gallery.UWP/Views/SamplePages/DataGridSamplePage.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+		public List<DataGridGridLinesVisibility> GridLinesVisibilityOptions { get => GetProperty<List<DataGridGridLinesVisibility>>(); set => SetProperty(value); }
+		public List<DataGridHeadersVisibility> HeadersVisibilityOptions { get => GetProperty<List<DataGridHeadersVisibility>>(); set => SetProperty(value); }
+		public List<DataGridSelectionMode> SelectionModeOptions { get => GetProperty<List<DataGridSelectionMode>>(); set => SetProperty(value); }
+
+		public string RowGroupHeader { get => GetProperty<string>(); set => SetProperty(value); }
+
+		public IEnumerable DataGridItems { get => GetProperty<IEnumerable>(); set => SetProperty(value); }
+		public FruitOrVegetableEnum[] FruitOrVegetableEnumValues;
+
+		// This is currently causes bug on WASM, removed until fixed
```
Comment (agneszitte) [inline Uno.Gallery/Uno.Gallery.UWP/Views/SamplePages/DataGridSamplePage.xaml.cs:177]:
```suggestion
		// This currently causes a bug on WASM, removed until fixed
```
Please add in the comment the related link for this specific issue

File: Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml [hunk 2]
```diff
@@ -0,0 +1,968 @@
...
+                            <!--<localprimitives:DataGridColumnHeader x:Name="TopRightCornerHeader" Grid.Column="2" Template="{StaticResource TopRightHeaderTemplate}"/>-->
+                            <Rectangle x:Name="ColumnHeadersAndRowsSeparator" Grid.ColumnSpan="3" VerticalAlignment="Bottom" Width="Auto" StrokeThickness="1" Height="1" Fill="{ThemeResource GridLinesBrush}"/>
+
+                            <localprimitives:DataGridRowsPresenter x:Name="RowsPresenter" Grid.ColumnSpan="3" Grid.Row="1" Grid.RowSpan="2"
+                                AutomationProperties.AccessibilityView="Raw"/>
+                            <Rectangle x:Name="BottomRightCorner" Fill="{ThemeResource ScrollBarsSeparatorBackground}" Opacity="0" Grid.Row="2" Grid.Column="2"/>
+                            <!-- Use BottomLeftCorner when RowsPresenter's Grid.RowSpan is 1 and the horizontal ScrollBar does not paint on top of the cells -->
+                            <!--<Rectangle x:Name="BottomLeftCorner" Fill="#FFE9EEF4" Grid.Row="2" Grid.ColumnSpan="2"/>-->
```
Comment (agneszitte) [inline Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml:409]: Related to an issue?

File: Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml [hunk 3]
```diff
@@ -0,0 +1,968 @@
...
+                            <Grid Grid.Column="1" Grid.Row="2">
+                                <Grid.ColumnDefinitions>
+                                    <ColumnDefinition Width="Auto"/>
+                                    <ColumnDefinition Width="*"/>
+                                </Grid.ColumnDefinitions>
+                                <Rectangle x:Name="FrozenColumnScrollBarSpacer"/>
+                                <ScrollBar x:Name="HorizontalScrollBar" Grid.Column="1" Orientation="Horizontal" Visibility="Collapsed"/>
+                            </Grid>
+
+                            <!--<dataInput:ValidationSummary x:Name="ValidationSummary" Grid.Row="3" Grid.ColumnSpan="3" MaxHeight="90"/>-->
```
Comment (agneszitte) [inline Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml:421]: Why is this commented ?

File: Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml [hunk 4]
```diff
@@ -0,0 +1,968 @@
...
+                        <VisualStateManager.VisualStateGroups>
+                            <VisualStateGroup x:Name="CommonStates">
+                                <VisualState x:Name="Normal"/>
+                                <VisualState x:Name="NormalCurrentRow">
+                                    <!-- For optional row currency and editing glyphs
+                                    <Storyboard>
+                                        <DoubleAnimation Storyboard.TargetName="CurrentRowGlyph" Storyboard.TargetProperty="Opacity" Duration="0" To="1"/>
+                                        <DoubleAnimation Storyboard.TargetName="EditingRowGlyph" Storyboard.TargetProperty="Opacity" Duration="0" To="0"/>
+                                    </Storyboard>
+                                    -->
```
Comment (agneszitte) [inline Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml:683]: commented for which reason?

File: Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml (hunk 2 above)
Comment (morning4coffe-dev) [reply Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml:409]: @agneszitte This one https://github.com/unoplatform/uno/issues/7635. This workaround still has to be used to make the DataGrid work on mobile targets.

File: Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml (hunk 3 above)
Comment (morning4coffe-dev) [reply Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml:421]: Same as for the https://github.com/unoplatform/Uno.Gallery/pull/705#discussion_r1196056414

File: Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml (hunk 4 above)
Comment (morning4coffe-dev) [reply Uno.Gallery/Uno.Gallery.UWP/Views/Styles/DataGrid.xaml:683]: Also same issue https://github.com/unoplatform/Uno.Gallery/pull/705#discussion_r1196056414

File: Uno.Gallery/Uno.Gallery.UWP/Views/SamplePages/DataGridSamplePage.xaml.cs (hunk 1 above)
Comment (morning4coffe-dev) [reply Uno.Gallery/Uno.Gallery.UWP/Views/SamplePages/DataGridSamplePage.xaml.cs]: Right, thanks! I forgot to add it, I have added it to the code. To link it with this pull request, it's this one https://github.com/unoplatform/uno/issues/11184

**Review by MartinZikmund — APPROVED** [review]

**Review by MartinZikmund — APPROVED** [review]

**Comment by MartinZikmund** [comment]
@morning4coffe-dev seems there are issues building this on iOS, can you check?

**Comment by morning4coffe-dev** [comment]
@MartinZikmund I will try to take a look, but I unfortunately still have no way to test it.

**Comment by MartinZikmund** [comment]
@morning4coffe-dev what is needed here?

**Comment by morning4coffe-dev** [comment]
@MartinZikmund 

- [ ] Resolve conflicts

- [ ] Update with the https://github.com/unoplatform/Uno.Gallery/pull/840 changes - the Android version should now work without any needed workarounds

- [ ] Fix it on iOS (this was my stop previously)

- [ ] Test on all platforms

**Comment by morning4coffe-dev** [comment]
Closing in favor of #1003

**Pull request closed without merging**